fili-core/src/main/java/com/yahoo/bard/webservice/logging/RequestLogUtils.java

```diff
@@ -0,0 +1,352 @@
+            }
+
+            timePhase = new TimedPhase(timePhaseName);
+            current.times.put(timePhaseName, timePhase);
+        }
+        current.mostRecentTimer = timePhase;
+        timePhase.start();
+    }
+
+    /**
+     * Stop the most recent stopwatch and start this one.
+     * Time is accumulated if the stopwatch is already registered.
+     *
+     * @param nextPhase  the name of the stopwatch to be started
+     */
+    public static void switchTiming(String nextPhase) {
+        stopMostRecentTimer();
+        startTiming(nextPhase);
+    }
+
+    /**
+     * Stops the request timer.
+     * Should be called exactly once, just before Fili responds to the request
+     */
+    public static void stopTimingRequest() {
+        stopTiming(TOTAL_TIMER);
+    }
+
+    /**
+     * Pause a stopwatch.
+     *
+     * @param caller  the caller to name this stopwatch with its class's simple name
+     */
+    public static void stopTiming(Object caller) {
+        stopTiming(caller.getClass().getSimpleName());
+    }
+
+    /**
+     * Pause a stopwatch.
+     *
+     * @param timePhaseName  the name of this stopwatch
+     */
+    public static void stopTiming(String timePhaseName) {
+        TimedPhase timePhase = RLOG.get().times.get(timePhaseName);
+        if (timePhase == null) {
+            LOG.warn("Tried to stop non-existent phase: {}", timePhaseName);
+            return;
+        }
+        timePhase.stop();
+    }
+
+    /**
+     * Stop the most recent stopwatch.
+     */
+    public static void stopMostRecentTimer() {
+        try {
+            stopTiming(RLOG.get().mostRecentTimer.name);
+        } catch (NullPointerException ignored) {
+            LOG.warn("Stopping timing failed because mostRecentTimer wasn't registered.");
+        }
+    }
+
+    /**
+     * Expose the request phases and their durations in nanoseconds.
+     *
+     * @return a mapping of phase -> duration (ns)
+     */
+    public static Map<String, Long> getDurations() {
+        return RLOG.get().getDurations();
+    }
+
+    /**
+     * Expose the request phases and their durations in milliseconds (including an entry for the longest druid query
+     * that ran).
+     *
+     * @return a mapping of phase -> duration (ms)
+     */
+    public static Map<String, Float> getAggregateDurations() {
+        return RLOG.get().getAggregateDurations(false);
+    }
+
+    /**
+     * Record logging information in the logging context.
+     *
+     * @param logPhase  the name of the class destined to hold this logging information
+     *
+     * @see LogBlock
+     */
+    public static void record(LogInfo logPhase) {
+        try {
+            RLOG.get().info.add(logPhase);
+        } catch (NullPointerException ignored) {
+            LOG.warn(
+                    "Attempted to append log info while request log object was uninitialized: {}",
+                    logPhase.getClass().getSimpleName()
+            );
+        }
+    }
```

Comment:
If we do merge `TimedPhase` and then this class then we'll probably want to move `record` back onto `RequestLog` (or have a `RequestLogUtils` class), because this really doesn't have anything to do with `TimedPhase`.

This applies to the rest of the methods in this class (I don't have a strong opinion one way or the other).